### PR TITLE
[cleanup] make error reporting more robust

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Add compiler warnings for `report_error()` and `report_info()`.

@dirkhh: Is there any point to these `SSRF_INFO` macros? They are just visual clutter to me. If you feel these are necessary, please drop the last commit.

### CODE REVIEW

**Comment by dirkhh** [comment]
As so often, `SSRF_INFO` has historical reasons. Initially this replaced `fprintf()` statements so that we could also write to the Android logger when building on Android. Of course this evolved over time and at this point you are correct, it's just a silly wrapper, so I'm good with simply dropping it.

File: core/errorhelper.h [hunk 1]
```diff
@@ -8,10 +8,18 @@
...
+#define __printf(x, y) __attribute__((__format__(__printf__, x, y)))
+#else
+#define __printf(x, y)
+#endif
+
 extern int verbose;
-extern int report_error(const char *fmt, ...);
+extern int __printf(1, 2) report_error(const char *fmt, ...);
+extern void __printf(1, 2) report_info(const char *fmt, ...);
 extern void set_error_cb(void(*cb)(char *));	// Callback takes ownership of passed string
```
Comment (dirkhh) [inline core/errorhelper.h:20]: so now we have no report_error declaration anymore? My quick search seems to indicate that errorhelper.h is the only place where it was declared in a header.
What am I missing?

**Review by dirkhh — COMMENTED** [review]
This all looks great and I was almost ready to merge it - and then I stumbled across the removal of the declaration which confused me...

File: core/errorhelper.h (hunk 1 above)
Comment (bstoeger) [reply core/errorhelper.h:20]: It's not removed. The `__printf(1,2)` attribute is added so that we get compiler checks.

Comment (dirkhh) [reply core/errorhelper.h:20]: and this is why you shouldn't do code reviews at 6 in the morning... duh

**Review by dirkhh — APPROVED** [review]
-ENOTENOUGHCOFFEE

**Pull request merged**